src/test/e2e-emulated/masterchain.spec.ts

```diff
@@ -84,13 +64,13 @@ describe("masterchain", () => {
             workchain: -1,
```

Comment:
I am curious: What would happen if this `workchain` property is not set, but a "DeployToMasterchain" message is sent to the contract?

Reply:
if `workchain` is, let's say `42`, then you'll see something like this in the tests:
```
  ● masterchain › should not deploy to workchain from wrong chain

    Error while executing transaction: Emulate transaction failed: [Error : -669 : cannot run message on account cannot create bounce phase of a new transaction for smart contract E8751B8F17DBDD7B2DDE96B9F303DDAC5B7137C7251097F01347B8466648138A]
    Blockchain logs:
    [ 4][t 0][2024-12-18 16:19:41.738000][transaction.cpp:443]  unpacking account E8751B8F17DBDD7B2DDE96B9F303DDAC5B7137C7251097F01347B8466648138A
    [ 4][t 0][2024-12-18 16:19:41.738000][transaction.cpp:274]  last_paid=0; cells=0 bits=0 public_cells=0
    [ 4][t 0][2024-12-18 16:19:41.738000][transaction.cpp:504]  end of Account.unpack() for 0:E8751B8F17DBDD7B2DDE96B9F303DDAC5B7137C7251097F01347B8466648138A (balance = 0ng ; last_trans_lt = 0..0)
    [ 4][t 0][2024-12-18 16:19:41.738000][transaction.cpp:3655] block random seed set to 0000000000000000000000000000000000000000000000000000000000000000
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1236] gas limits: max=1000000, limit=1000000, credit=0
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1571] HASH(in_msg_state) = E8751B8F17DBDD7B2DDE96B9F303DDAC5B7137C7251097F01347B8466648138A, account_state_hash = E8751B8F17DBDD7B2DDE96B9F303DDAC5B7137C7251097F01347B8466648138A
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1645] creating VM
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1351] SmartContractInfo initialized with [ 124711402 0 0 1734538781 4000000 4000000 58705890321835810842207472787228479690183528187301183539049416341802289973577 [ 10000000000 (null) ] CS{Cell{005fc00e8751b8f17dbdd7b2dde96b9f303ddac5b7137c7251097f01347b8466648138a00000000000000000000000000004} bits: 1..268; refs: 0..0} C{A7EC01478F770EF8F8B037CC7E42F0ABF0D85A50E1828C2394DAD1411691F2DE} C{414F4C6E823A0E565520EBE214CD2C17A95A2D22CB2E8D2C6FCCBE5D414A5008} [ 10000000000 (null) ] 0 (null) [ CS{Cell{004dd06600000000000000000000000080000000000000fa00000000000001f4000000000003d09040} bits: 9..305; refs: 0..0} (null) CS{Cell{0094d1000000000000006400000000000f4240de000000002710000000000000000f42400000000002160ec0000000000000271000000000002625a00000000005f5e100000000003b9aca00} bits: 0..592; refs: 0..0} CS{Cell{0094d100000000000000640000000000009c40de000000000190000000000000000f424000000000000f4240000000000000271000000000009896800000000005f5e100000000003b9aca00} bits: 0..592; refs: 0..0} CS{Cell{0042ea000000000098968000000000271000000000000f424000000001800055555555} bits: 0..264; refs: 0..0} CS{Cell{0042ea0000000000061a800000000001900000000000009c4000000001800055555555} bits: 0..264; refs: 0..0} (null) ] 0 (null) ]
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1675] starting VM
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1680] VM terminated with exit code 1136
    [ 3][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1702] steps: 80 gas: used=3229, max=1000000, limit=1000000, credit=0
    [ 3][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1704] out_of_gas=false, accepted=true, success=false, time=0.000306s
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:1737] gas fees: 1291600 = 26214400 * 3229 /2^16 ; price=26214400; flat rate=[40000 for 100]; remaining balance=9998708400ng
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:2194] destination address contains unknown workchain_id 42
    [ 4][t 0][2024-12-18 16:19:41.739000][transaction.cpp:2874] invalid destination address in a bounced message

      90 |         );
      91 |
    > 92 |         const deployResult = await contract.send(
         |                              ^
      93 |             treasure.getSender(),
      94 |             { value: toNano("10") },
      95 |             "DeployToWorkchain",

      at SmartContract.runCommon (node_modules/@ton/sandbox/dist/blockchain/SmartContract.js:234:19)
      at SmartContract.receiveMessage (node_modules/@ton/sandbox/dist/blockchain/SmartContract.js:213:16)
      at Blockchain.processInternal (node_modules/@ton/sandbox/dist/blockchain/Blockchain.js:267:22)
      at Blockchain.processTx (node_modules/@ton/sandbox/dist/blockchain/Blockchain.js:312:96)
      at node_modules/@ton/sandbox/dist/blockchain/Blockchain.js:323:30
      at AsyncLock.with (node_modules/@ton/sandbox/dist/utils/AsyncLock.js:40:20)
      at Blockchain.processQueue (node_modules/@ton/sandbox/dist/blockchain/Blockchain.js:319:16)
      at Blockchain.runQueue (node_modules/@ton/sandbox/dist/blockchain/Blockchain.js:244:22)
      at Proxy.<anonymous> (node_modules/@ton/sandbox/dist/blockchain/Blockchain.js:455:40)
      at Object.<anonymous> (src/test/e2e-emulated/masterchain.spec.ts:92:30)
```